### PR TITLE
[template] Update instructions

### PR DESCRIPTION
**Pull request by llvm-beanz**
There's been a bunch of confusion around this process, so I figured updating the instructions to be a bit more explicit might help.

### CODE REVIEW

**Review by coopp — APPROVED** [review]
Looks good to me!. --- incoming PR for adding a proper number to my recent proposal :)

File: proposals/templates/basic-template.md [hunk 1]
```diff
@@ -11,10 +11,19 @@
...
-> 2. Do not spend time writing the "Detailed design" until the feature has been
->    merged in the "Under Consideration" phase.
-> 3. Delete this Instructions section including the line below.
+> 1. Copy this template to proposals and name it `NNNN-<some unique name>.md`.
+>    Do not assign the proposal a number.
+> 2. Exclude the "Planned Version", "PRs" and "Issues" from the header as
+>    appropriate.
+> 3. Fill out the "Introduction", "Motivation", and "Proposed Solution" sections
+>    of the template. Do not spend time writing the "Detailed design" until the
+>    feature has been merged in the "Under Consideration" phase.
```
Comment (damyanp) [inline proposals/templates/basic-template.md:20]: Note that there is nothing in this repo describing the different phases of proposals.

**Review by damyanp — APPROVED** [review]

File: proposals/templates/basic-template.md (hunk 1 above)
Comment (llvm-beanz) [reply proposals/templates/basic-template.md:20]: Good point. Maybe the phases just don't matter for these so we should remove it from the template entirely.

Comment (damyanp) [reply proposals/templates/basic-template.md:20]: I think they may still be useful.  For example, the root signature proposal in here is not complete and there isn't consensus that it's the right way to go, so it'd be good if we could capture that somehow.

**Pull request merged**